lib/vdsm/storage/mpathconf.py

```diff
@@ -428,3 +429,26 @@ def read_metadata():
         return Metadata(REVISION_OLD, private)
 
     return Metadata(REVISION_MISSING, private)
+
+
+def is_ufn_disabled():
+    """
+    Check if 'user_friendly_names' is enabled in the multipathd configuration.
+    The configuration is obtained by running the multipathd command to show
+    the local configuration and capturing the output.
+    Parse the output and check that the contents are valid.
+
+    Returns:
+        List[str]: contains the names of the offending sections
+            (had 'user_friendly_names' enabled). Empty otherwise.
+    """
+    # Assumptions in the regex:
+    # - If a section has nested sections, will not match key-value pairs in
+    #   the top level, only their sub-sections have key-value pairs checked.
+    # - Values can be either strings, numerical values, or quoted strings.
+    #   Therefore, match enabling strings both with and without double quotes.
+    # - Between the section string and the left bracket, a single space is
+    #   allowed as maximum.
+    # - Between keys and values at least one space is required.
+    pattern = r'\s+(\S+) +{[^{]+user_friendly_names +"?yes"?\s+}'
```

Comment:
Few issues:
- The section name always starts with leading whitespace - how does it work for

      defaults {

      }

- This assumes that "user_friendly_names yes" is the last line before the "}"

This would be more clear if we use named regex and verbose option:

Reply:
I think I tried key-values after the "user_friendly_names yes" and worked, but you are right that it should only accept "blank" characters, so maybe I didn't test it right.

I am currently trying to clear a few of the assumptions to rework the regex, but I'll take these comments into consideration next time I push. Possibly also add some simple tests to increase the coverage.

I'll draft the PR for now, until I figure out the small opens.